base/steer/FairRootManager.cxx

```diff
@@ -974,70 +978,81 @@ void FairRootManager::DeleteOldWriteoutBufferData()
     }
 }
 
-char* FairRootManager::GetTreeName()
+std::string FairRootManager::GetNameFromFile(const char* namekind)
 {
-    char* default_name = (char*)"cbmsim";
+    std::string default_name = "";
+
+    if (strcmp(namekind, "treename=") && strcmp(namekind, "foldername=")) {
+        LOG(fatal) << "FairRootManager, requested " << namekind << ", while only treename= and foldername= available.";
+        return default_name;
+    }
+
     char* workdir = getenv("VMCWORKDIR");
     if (nullptr == workdir) {
         return default_name;
     }
 
     // Open file with output tree name
-    FILE* file = fopen(Form("%s/config/rootmanager.dat", workdir), "r");
+    std::ifstream file(Form("%s/config/rootmanager.dat", workdir));
     // If file does not exist -> default
-    if (nullptr == file) {
+    if (!file.is_open()) {
         return default_name;
     }
 
-    char str[100];
-    while (nullptr != fgets(str, 100, file)) {
-        if (TString(str).Contains("treename")) {
-            char* treename = new char[100];
-            if (1 == sscanf(str, "treename=%s", treename)) {
-                fclose(file);
-                return treename;
-            }
-            delete[] treename;
+    std::string line;
+    while (std::getline(file, line)) {
+        if (line.rfind(namekind, 0) == 0) {
+            line.replace(0, strlen(namekind), "");
+            file.close();
+            return line;
         }
     }
 
-    // Key "treename" was not found in file: return default
-    fclose(file);
+    file.close();
     return default_name;
 }
 
-char* FairRootManager::GetFolderName()
+char* FairRootManager::GetTreeName()
 {
-    char* default_name = (char*)"cbmout";
-    if (!FairRun::Instance()->IsAna()) {
-        default_name = (char*)"cbmroot";
+    if (fTreeName.length() > 0) {
+        char* treeName = new char[fTreeName.length()];
+        strcpy(treeName, fTreeName.c_str());
+        return treeName;
     }
-    char* workdir = getenv("VMCWORKDIR");
-    if (nullptr == workdir) {
-        return default_name;
+
+    std::string nameFromFile = GetNameFromFile("treename=");
+    if (nameFromFile.length() > 0) {
+        char* treeName = new char[nameFromFile.length()];
```

Comment:
You have to allocate `fTreeName.length() + 1`. The final NUL byte needs to have space as well.